### PR TITLE
docs: add examples for index.ts files on server routes

### PR DESCRIPTION
**Pull request by filipweidemann**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

While getting to know how Nuxt handles server routes I stumbled over the required naming of the `.ts` files to create a rather common API structure:

* `/api/project/`
* `/api/project/[id]`
* `/api/project/[id]/invite`

The docs didn't convey that handling the "root" namespace of you API requires setting one (or multiple) `index.ts` files inside of the `project` directory so that `/api/project` will be handled correctly and with potentially multiple HTTP verbs and different logic.

This is obviously a quite common technique in all kinds of module-situations in the JS world but I thought it would be cool to include this as a short and concise example alongside the existing docs about the file router.

I am also aware that many things are more or less directly related to `nitro` and may or may not need to be directly documented inside the Nuxt docs but since this documentation is the first contact point for everyone learning Nuxt I though that I could help to make this more clear for all future learners!

If you have any question or opinion about this, please ask straight away. I'd love to make this even better if needed. :)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/2.guide/2.directory-structure/1.server.md [hunk 1]
```diff
@@ -180,6 +180,38 @@ Given the example above, fetching `/test` with:
 - **POST** method: Returns `Test post handler`
 - Any other method: Returns 405 error
 
+You may also name files like `index.[verb].ts` if you want to compose your routes inside different folders for better readability.
```
Comment (Atinux) [inline docs/2.guide/2.directory-structure/1.server.md:183]:
```suggestion
You can also use `index.[method].ts` inside a directory for structuring your code differently.
```

File: docs/2.guide/2.directory-structure/1.server.md [hunk 2]
```diff
@@ -180,6 +180,38 @@ Given the example above, fetching `/test` with:
...
+
+**Example:**
+
+```ts [server/api/foo/index.ts]
+export default defineEventHandler((event) => => {
+  // handle the `api/foo` endpoint
+})
+```
+
+This is especially powerful if you want to enable different logic based on your HTTP verbs on the root of the API namespace as well as other logic that should get own routes inside the `foo` namespace:
```
Comment (Atinux) [inline docs/2.guide/2.directory-structure/1.server.md:193]:
```suggestion
This is useful to create API namespaces.
```

**Comment by filipweidemann** [comment]
Thanks @Atinux. I committed your suggestions.

**Review by atinux — APPROVED** [review]

File: docs/2.guide/2.directory-structure/1.server.md [hunk 3]
```diff
@@ -180,6 +180,38 @@ Given the example above, fetching `/test` with:
 - **POST** method: Returns `Test post handler`
 - Any other method: Returns 405 error
 
+You can also use `index.[method].ts` inside a directory for structuring your code differently.
+
+**Example:**
+
+```ts [server/api/foo/index.ts]
+export default defineEventHandler((event) => => {
```
Comment (DamianGlowala) [inline docs/2.guide/2.directory-structure/1.server.md:188]: @Atinux all examples contain a duplicated arrow

**Pull request merged**